### PR TITLE
Added execution and state modules for Vault

### PR DESCRIPTION
**Pull request by blarghmatey**
In order to make it easier to manage the Vault service I added an execution and state module to expose the capabilities of the `HVAC` library. Some of the current functionality being used in that library is only present in our fork of it (mitodl/hvac) until it gets merged into upstream.

### CODE REVIEW

**Comment by blarghmatey** [comment]
@bdero This is ready for review now

File: extensions/_modules/vault.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    pass
+
+def __virtual__():
+    return DEPS_INSTALLED
+
+
+def _build_client(url='https://localhost:8200', token=None, cert=None,
+                  verify=True, timeout=30, proxies=None, allow_redirects=True,
+                  session=None):
+    client_kwargs = locals()
```
Comment (bdero) [inline extensions/_modules/vault.py:46]: This seems ok. Locals can never include anything besides the kwargs here, right?

Comment (blarghmatey) [inline extensions/_modules/vault.py:46]: So far as I have been able to determine that is correct.

File: extensions/_states/vault.py [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+           'result': '',
+           'changes': {}}
+    initialized = __salt__['vault.is_initialized']()
+
+    if initialized:
+        ret['result'] = True
+        ret['Comment'] = 'Vault is already initialized'
+    elif __opts__['test']:
+        ret['result'] = None
+        ret['comment'] = 'Vault will be initialized.'
```
Comment (bdero) [inline extensions/_states/vault.py:36]: I recall seeing checks like this in salt (and it seems **opts** just contains the [master/minion conf settings](https://docs.saltstack.com/en/latest/topics/development/dunder_dictionaries.html)?), but what's the reason for changing this behavior when `test` is set?

File: extensions/_states/vault.py [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+            'root_credentials': {
+                'new': {
+                    'sealing_keys': sealing_keys,
+                    'root_token': root_token
+                },
+                'old': {}
+            }
+        }
+        ret['comment'] = 'Vault has {}initialized'.format(
+            '' if success else 'not ')
```
Comment (bdero) [inline extensions/_states/vault.py]: 'failed to ' might make this message a little more clear

File: extensions/_states/vault.py (hunk 2 above)
Comment (blarghmatey) [inline extensions/_states/vault.py:36]: https://docs.saltstack.com/en/latest/ref/states/writing.html#test-state

Comment (bdero) [inline extensions/_states/vault.py:36]: Thanks, that clears it up

**Comment by bdero** [comment]
I only have minor questions/comments :+1:

**Pull request merged**